cucumber-java/src/main/java/io/cucumber/java/MethodScanner.java

```diff
@@ -51,8 +51,13 @@ private static boolean isInstantiable(Class<?> clazz) {
     }
 
     private static void scan(BiConsumer<Method, Annotation> consumer, Class<?> aClass, Method method) {
-        // prevent unnecessary checking of Object methods
-        if (Object.class.equals(method.getDeclaringClass())) {
+        // prevent unnecessary checking of Object methods and generated methods
+        // Note on generated methods: when a class implements a method from the
+        // interface but specializes the return type, two Method will be
+        // generated (one with the return type of the interface and one with
+        // the specialized return type). The one with the return type of the
+        // interface is a synthetic one.
+        if (Object.class.equals(method.getDeclaringClass()) || method.isSynthetic()) {
```

Comment:
Please split this condition into two statements. It will make the code comments easier to read.